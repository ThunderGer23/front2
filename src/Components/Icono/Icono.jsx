import PropTypes from 'prop-types'

export const Icono = ({texto,Direccion}) => {
  return (
    <>
    <p role='p'>{texto}</p>
    <img role='img' className="logo" src={Direccion}  width="45"/>
    </>
  )
}

Icono.propTypes = {
    texto: PropTypes.string.isRequired,
    Direccion: PropTypes.string.isRequired
}