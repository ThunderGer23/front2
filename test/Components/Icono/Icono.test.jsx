import { render, screen } from "@testing-library/react";
import { Icono } from "../../../src/Components/Icono/Icono";

describe('Pruebas en el componente Icono', () => { 
    const texto='Aten'
    const Direccion = '../../../public/img/mq16.png'

    test('Esta prueba es para hacer matchcon el snapshot', () => { 
        const {container} = render(<Icono texto={texto} Direccion={Direccion}/>)//Renderiza el componente Icono
        expect(container).toMatchSnapshot()//Toma una foto
     })

    test('Esta prueba valida que el componente p exista', () => { 
        render(<Icono texto={texto} Direccion={Direccion}/>)
        expect(screen.findByRole('p')).toBeTruthy()//Se verifica que exista el p
     })

    test('Esta prueba valida que el componente img exista y tenga la clase correcta', () => { 
        render(<Icono texto={texto} Direccion={Direccion}/>)
        expect(screen.findByRole('img')).toBeTruthy()
        const {className} = screen.getByRole('img')//Se salva la clase de la img del componente en className
        expect(className).toBe('logo')
     })
 })