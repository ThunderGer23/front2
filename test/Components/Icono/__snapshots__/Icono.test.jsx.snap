// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Icono Esta prueba es para hacer matchcon el snapshot 1`] = `
<div>
  <p
    role="p"
  >
    Aten
  </p>
  <img
    class="logo"
    role="img"
    src="../../../public/img/mq16.png"
    width="45"
  />
</div>
`;
